SpecsUtils/src/pt/up/fe/specs/util/treenode/ATreeNode.java

```diff
@@ -68,47 +66,40 @@ private List<K> initChildren(Collection<? extends K> children) {
         return new ArrayList<>(children.size());
     }
 
-    /* (non-Javadoc)
+    /*
+     * (non-Javadoc)
+     * 
      * @see pt.up.fe.specs.util.treenode.TreeNode#getChildren()
      */
     @Override
     public List<K> getChildren() {
-        return this.children;
-
-        // Currently cannot enforce immutable children view due to MATISSE passes that directly modify children
-        // return Collections.unmodifiableList(this.children);
+        return Collections.unmodifiableList(this.children);
```

Comment:
Take a look at this before merging

Reply:
This is a change I think should be made but breaks `ATreeNode.removeChildren()`.